http-client/src/test/java/io/airlift/http/client/AbstractHttpClientTest.java

```diff
@@ -511,6 +520,70 @@ public void testPutMethodWithFileBodyGenerator()
         assertTrue(testFile.delete());
     }
 
+    @Test
+    public void testPutMethodWithStreamingBodyGenerator()
```

Comment:
I don't fully understand this test tbh. Please explain

Reply:
It tests the response stream can be outside after `httpClient.execute()` returns. Note that prior to this change, `httpClient.execute()` always closed the response stream and ended the request. This test proves that the response stream still works correctly, etc. after `httpClient.execute()`.

Reply:
This test proves piping the result of one request as the body of another request, which is the main purpose of this change. For example, imagine a proxy that sits between a client and a server. The client, say, generates 32GB of data for the remote server. Without this change, the proxy would need to spool the 32GB somewhere before sending to the remote client. This test proves that this works. It executes a request and, in the response handler, executes _another_ request while piping the response of the first request as the body of the second request.

Reply:
Update - I took a look again and see a big mistake. Let me correct it...

Reply:
It's better now. Sorry about that.

Reply:
Maybe let's pipe from tmp file instead and on the servlet side assert on the received length?

Reply:
Let's do both. I really want to test an existing, open request streaming into another request as that's the main use-case.